native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -118,30 +119,68 @@ public static Object convertToJSON(BXml xml, String attributePrefix, boolean pre
      * @return ObjectNode Json object node corresponding to the given xml element
      */
     private static Object convertElement(BXmlItem xmlItem, String attributePrefix,
-                                         boolean preserveNamespaces) {
+                                         boolean preserveNamespaces, AttributeManager attributeManager) {
         BMap<BString, Object> rootNode = newJsonMap();
-        LinkedHashMap<String, String> attributeMap = collectAttributesAndNamespaces(xmlItem, preserveNamespaces);
+        BMap<BString, Object> mapData = newJsonMap();
+        if (preserveNamespaces) {
+            getAttributes(xmlItem, preserveNamespaces, attributePrefix, mapData, attributeManager);
+        }
         String keyValue = getElementKey(xmlItem, preserveNamespaces);
-        Object children = convertBXmlSequence((BXmlSequence) xmlItem.getChildrenSeq(),
-                attributePrefix, preserveNamespaces);
-
-        if (children == null) {
-            if (attributeMap.isEmpty()) {
-                putAsBStrings(rootNode, keyValue, "");
+        Object children = convertBXmlSequence((BXmlSequence) xmlItem.getChildrenSeq(), attributePrefix,
+                preserveNamespaces, attributeManager);
+        if (mapData.size() > 0) {
+            if (children == null || children instanceof BString) {
+                if (children != null) {
+                    putAsBStrings(mapData, CONTENT, children.toString().trim());
+                    rootNode = mapData;
+                } else {
+                    putAsBStrings(rootNode, keyValue, mapData);
+                }
             } else {
-                BMap<BString, Object> attrMap = newJsonMap();
-                addAttributes(attrMap, attributePrefix, attributeMap);
-
-                putAsBStrings(rootNode, keyValue, attrMap);
+                BMap<BString, Object> data = (BMap<BString, Object>) children;
+                for (Map.Entry<BString, Object> entry: mapData.entrySet()) {
+                    data.put(entry.getKey(), entry.getValue());
+                }
+                putAsBStrings(rootNode, keyValue, data);
+            }
+        } else {
```

Comment:
Again will the element be considered XMLSequence without the inner tags greater than or equal to zero?

Reply:
Ack